### DISCUSSION TITLE
Adding a word to the personal dictionary unexpectedly exits jinx mode

### DISCUSSION THREAD

**Discussion by shipmints** [minad/jinx#203] [Q&A]
I expected to be able to add a new word to the dictionary and keep on spelling. Perhaps my settings need adjustment, or my expectations. Emacs 29.3 with vertico.

**Comment by minad** [comment]
I cannot reproduce this. With `C-u M-$` the whole buffer is spell checked. Jinx iterates over all misspellings and adding to the personal dictionary doesn't affect this behavior.

**Reply by shipmints** [reply]
I was using jinx-correct without a prefix argument to check visible words starting with the "nearest" which is what it says on the tin.

**Reply by minad** [reply]
`jinx-correct` is a DWIM dispatcher command, see the docstring. Without a prefix it only corrects the nearest word and exits after this single word. If you want to check more use `C-u M-$`, or you could also mark a region to check. It is also easy to hack your own commands based on the existing ones if you prefer a slightly different UI.

**Reply by shipmints** [reply]
I see. Then this might be a bug. If you jinx-correct and then type M-n, it is bound to jinx-next in that mode yet doesn't behave like jinx-next when checking a region/buffer. Or perhaps it would be easy to make it assume the whole buffer when jinx-next is invoked?

**Reply by minad** [reply]
Can you please be very precise about your expectations, since I am not sure if I understood you?

I just tested the behavior again. When I select a region press M-$ or when I press C-u M-$ for whole buffer check, then all the misspellings in the region or the buffer are iterated over. Furthermore M-n/M-p can be used to step over the misspellings. The ones outside the region are excluded. If you press M-$ without a region, only the nearest word is correct. You can also jump to the next misspelling, but only misspellings within the visible window region are considered. So the behavior is different in all the cases. The reason is that forcing the check of all misspellings in a whole large buffer can be expensive. Therefore M-$ (or M-$ on a region) only considers the misspellings which are present there. Does this sound reasonable?
